fbw-a380x/src/systems/instruments/src/MFD/pages/common/Button.tsx

```diff
@@ -0,0 +1,256 @@
+          while (this.dropdownMenuRef.instance.firstChild) {
+            this.dropdownMenuRef.instance.removeChild(this.dropdownMenuRef.instance.firstChild);
+          }
+
+          this.renderedMenuItems = items;
+
+          // Render dropdownMenuRef's children
+          const itemNodes: VNode = (
+            <div>
+              {items?.map<VNode>(
+                (el, idx) => (
+                  <span id={`${this.props.idPrefix}_${idx}`} class="mfd-dropdown-menu-element">
+                    {el.label}
+                  </span>
+                ),
+                this,
+              )}
+            </div>
+          );
+          FSComponent.render(itemNodes, this.dropdownMenuRef.instance);
+
+          // Add click event listener
+          items?.forEach((val, i) => {
+            document.getElementById(`${this.props.idPrefix}_${i}`)?.addEventListener('click', () => {
+              val.action();
+              this.dropdownIsOpened.set(false);
+            });
+          });
+
+          // Check if menu would overflow vertically (i.e. leave screen at the bottom). If so, open menu upwards
+          // Open menu for a split second to measure size
+          this.dropdownMenuRef.instance.style.display = 'block';
+          this.buttonRef.instance.classList.add('opened');
+
+          // Check if menu leaves screen at the bottom, reposition if needed
+          const boundingRect = this.dropdownMenuRef.instance.getBoundingClientRect();
+          const overflowsVertically = boundingRect.top + boundingRect.height > 1024;
+          this.menuOpensUpwards.set(overflowsVertically);
+
+          if (overflowsVertically === true) {
+            this.dropdownMenuRef.instance.style.top = `${Math.round(-boundingRect.height)}px`;
+          }
+
+          // Close again
+          this.dropdownMenuRef.instance.style.display = 'none';
+          this.buttonRef.instance.classList.remove('opened');
+        }, true),
+      );
+    }
+
+    this.subs.push(
+      this.props.label?.sub((val) => {
+        while (this.buttonRef.instance.firstChild) {
+          this.buttonRef.instance.removeChild(this.buttonRef.instance.firstChild);
+        }
+
+        // If menuItems is defined, render as button with arrow on the right side
+        if (this.props.menuItems !== undefined && this.props.showArrow === true) {
+          const n: VNode = (
+            <div class="mfd-fms-fpln-button-dropdown">
+              <span class="mfd-fms-fpln-button-dropdown-label">{val}</span>
+              <span class="mfd-fms-fpln-button-dropdown-arrow">
+                {this.menuOpensUpwards.get() ? (
+                  <TriangleUp color={this.props.disabled?.get() ? 'grey' : 'white'} />
+                ) : (
+                  <TriangleDown color={this.props.disabled?.get() ? 'grey' : 'white'} />
+                )}
+              </span>
+            </div>
+          );
+          FSComponent.render(n, this.buttonRef.instance);
```

Comment:
I am not a fan of these explicit calls to `FSComponent.render`, but I don't know if there's a better way to do it.

Reply:
Haven't found one :/